### ISSUE TITLE
Relaxe the rule against `using namespace` inside another namespace

### ISSUE THREAD

**Issue by fwyzard** [cms-sw/cmssw#40322]
While testing #40285 the CI came up with this error

> Rule1 : https://raw.githubusercontent.com/cms-sw/cmssw/master/Utilities/ReleaseScripts/python/cmsCodeRules/config.py
> Search for "using namespace" or "using std::" in header files
> 
> /DataFormats/PortableTestObjects/interface/alpaka/TestDeviceCollection.h

coming from this part of the code:
```c++
namespace ALPAKA_ACCELERATOR_NAMESPACE {

  namespace portabletest {

    // make the names from the top-level portabletest namespace visible for unqualified lookup
    // inside the ALPAKA_ACCELERATOR_NAMESPACE::portabletest namespace
    using namespace ::portabletest;

```

I think "importing" a `namespace` in a different, non-top-level `namespace` should be allowed, and the rule and the check should be relaxed.

**Comment by fwyzard** [comment]
assign core

**Comment by smuzaffar** [comment]
@fwyzard ,  I agree. Currently this rule is relaxed for `comments` and `functions` . We should allow to skip it for `namespaces` too. I will check if there is any clang-tidy check which can impose this rule otherwise we need to add `cppNamespaceSkppier` for our rule checker

**Comment by makortel** [comment]
We already have a check in static analyzer for `using namespace` in headers, and by looking at the result, it seems to be working as expected. So one option would be to leverage that, maybe in somewhat similar way as we did at one point in time to flag the `EventSetupRecord::get()` calls (before we moved to collecting deprecation warnings from compilation log).

An alternative that came to my mind would be to keep this check, but exploit the fact that any `using namespace` inside a namespace or e.g. function scope gets intended by `clang-format`. So it _might_ be sufficient to simply change the checker to require the `using` is in the beginning of the line.

**Comment by makortel** [comment]
@smuzaffar Should we
* Drop the specific `using namespace` check and rely on the static analyzer check as it is
* Drop the specific `using namespace` check, and make the PR test summary message to flag the `using namespace` warnings
* Improve the check to flag only those `using namespace` that add stuff into the global namespace?

**Comment by smuzaffar** [comment]
@makortel , I am happy to drop the `using namespace` check and rely on the static analyzer check but are we sure the static analyzer is working fine in this case? Let me open a PR to test this

**Comment by makortel** [comment]
> I am happy to drop the `using namespace` check and rely on the static analyzer check but are we sure the static analyzer is working fine in this case? Let me open a PR to test this

I'd expect it to, but good to test. A downside would be that the static analyzer warnings do not (presently) propagate to the PR summary test, whereas the code rule check results do.

**Comment by smuzaffar** [comment]
Looks like our static analyzer check is not working. I opened https://github.com/cms-sw/cmssw/pull/46550 and static analyzer did not dump any warnings https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4f917a/42473/clang-logs/ https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-4f917a/42473/llvm-analysis/ .  Looks like https://github.com/cms-sw/cmssw/blob/master/Utilities/StaticAnalyzers/src/CmsSupport.cpp#L33-L44 does not work properly when we build with relative path ( src/package/filename instead of $LOCALRT/src/package/filename). I will provide a fix
